### DISCUSSION TITLE
Hyprland에서 kime을 올바르게 설정하는 방법 (kitty 정상작동)

### DISCUSSION THREAD

**Discussion by seolcu** [Riey/kime#653] [Show and tell]
> 2025-05-05 수정: [uwsm](https://wiki.hyprland.org/Useful-Utilities/Systemd-start/) 관련 설명 추가, 한글키 바인딩 제거

Hyprland를 설치 후 kime을 설정해보았는데 kitty에서 한글 입력이 안되더라구요.
그래서 이슈를 뒤지던 중 이걸 찾았습니다. (https://github.com/Riey/kime/issues/647)
이를 바탕으로 올바른 설정법을 찾아냈습니다.

### 1-(1). [uwsm](https://wiki.hyprland.org/Useful-Utilities/Systemd-start/) 사용시

`~/.config/uwsm/env`를 편한 에디터로 열어서 다음 내용을 넣어주세요.

```
export GTK_IM_MODULE=kime
export QT_IM_MODULE=kime
export XMODIFIERS=@im=kime
```

[uwsm](https://wiki.hyprland.org/Useful-Utilities/Systemd-start/)은 autostart를 지원하므로 재부팅하면 바로 정상적으로 실행됩니다.

### 1-(2). [uwsm](https://wiki.hyprland.org/Useful-Utilities/Systemd-start/) 미사용시

`~/.config/hypr/hyprland.conf`를 편한 에디터로 열어주세요.

파일의 시작 부분에 다음을 추가해주세요.

```
# Kime 실행
exec-once = kime

# Kime 환경변수 설정
env = GTK_IM_MODULE,kime
env = QT_IM_MODULE,kime
env = XMODIFIERS,@im=kime
```

재부팅 또는 로그아웃 후 로그인하면 사용가능합니다.

## 2. 한글 & 한자기 설정

`~/.config/hypr/hyprland.conf`를 편한 에디터로 열어주세요.

조금 아래로 내려보시면 input이 보일 텐데, 중괄호 안에 kb_options =가 있습니다. 그 부분을 다음과 같이 바꿔주세요.

```
    # korean input with ralt & rctrl
    kb_options = korean:ralt_hangul,korean:rctrl_hanja
```

#### 참고한 글

- [Hyprland Wiki uwsm](https://wiki.hyprland.org/Useful-Utilities/Systemd-start/)
- [참고한 이슈](https://github.com/Riey/kime/issues/647)
- [Hyprland 환경변수 설정](https://wiki.hyprland.org/Configuring/Environment-variables/)
- [Hyprland 키보드 옵션](https://wiki.hyprland.org/Configuring/Variables/)

**Comment by imyme010101** [comment]
안녕하세요. 
잘 작동 합니다. 다만 제가 설정을 잘못했는지는 모르겠지만 파이어폭스 kitty 동시에 켜두었을떄 한쪽에서 키가 안먹는 현상이 있는데 
이문제 해결법 혹시 아실까요?

**Reply by seolcu** [reply]
죄송합니다ㅠㅠ 답을 너무 늦게 봤네요. 저는 이런 문제가 없었기 때문에 명확히 말씀드리기 어렵지만, 설정하신 hyprland config쪽에 문제가 있는 것 같네요.

**Comment by klfnn** [comment]
감사합니다ㅠㅠㅠ 너무 잘 작동해요!

**Reply by seolcu** [reply]
도움이 되었다니 너무 기쁘네요~ 잘 사용하세요!
